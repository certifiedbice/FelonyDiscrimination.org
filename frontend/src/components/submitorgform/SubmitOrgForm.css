#submit-form-instructions {
    padding: 0 20px 20px 20px;
}

/* Desktop CSS */
@media(min-width:991px){
    
    #submit-form-instructions {
        padding: 0 60px 20px 60px;
    }
    
}