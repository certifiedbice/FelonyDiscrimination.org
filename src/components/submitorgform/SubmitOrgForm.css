#submit-form-container {
    flex: 1;
    text-align: center;
}

#submit-form {}

#submit-form legend {
    width: 100%;
    margin-bottom: 5px;
    font-size: 18px;
    font-weight: 700;
}

#submit-form-instructions {
    padding: 0 20px 20px 20px;
}

.submit-form-element-container {
    padding: 5px;   
}

.submit-form-element-container label {
    display: block;
    margin: 0 0 5px 20px;
    text-align: left;
    font-size: 12px;
}

.submit-form-element {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

.submit-form-element-container input {
    border-radius: 50px;
    padding: 5px;
}

.submit-form-element-container select {
    width: 95%;
    border-radius: 50px;
    padding: 5px;
}

.submit-form-element-container textarea {
    border-radius: 25px;
    padding: 10px;
}

#submit-form-submit {
    width: 90%;
    background-color: #4378bc;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
    cursor: pointer;
}

#submit-form-submit:hover {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

/* Desktop CSS */
@media(min-width:991px){
    
    #submit-form-instructions {
        padding: 0 60px 20px 60px;
    }
    
    .submit-form-element-container label {
        margin: 0 0 10px 10px;
    }
    
    .submit-form-element {
        width: 100%;
        background-color: #323232;
        border: 1px solid #e6e7e8;
        font-size: 12px;
        color: #e6e7e8;
    }

    .submit-form-element-container select {
        width: 100.5%;
    }
    
    #submit-form-submit {
        width: 25%;
    }

    #submit-form-submit:hover {
        width: 25%;
    }

}