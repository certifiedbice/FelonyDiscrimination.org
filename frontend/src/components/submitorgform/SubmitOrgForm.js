import React,{Component} from 'react';
import {Link} from 'react-router-dom';
import OrgApiService from '../../services/org-api-service';
import TokenService from '../../services/token-service';
import './SubmitOrgForm.css';

export default class SubmitOrgForm extends Component{
	
	state={error:null};
	
	renderSubmitOrgForm(){
		return(
			<form id='submit-org-form' name='submit-org-form' aria-label='submit-org-form' onSubmit={this.handleSubmit}>
				<fieldset>
					<legend><h2 id='submit-org-h2'>Submit an Organization</h2></legend>
					<div id='submit-org-form-instructions'>
						<p>We invite you to submit organizations into the database, where they can then be reviewed as positive or negative for discrimination.</p>
						<p>We would like to thank you for taking the time to submit an entry, every entry is one step closer to a solution.</p>
					</div>
					<div className='form-element-container'>
						<label htmlFor='submit-org-form-organization-name'>Organization name:</label>
						<input id='submit-org-form-organization-name' name='org_name' type='text' required aria-labelledby='submit-org-form-organization-name' placeholder='Name'/>
					</div>
					<div className='form-element-container'>
						<label htmlFor='submit-org-form-organization-phone'>Phone:</label>
						<input id='submit-org-form-organization-phone' name='org_phone' type='tel' aria-labelledby='submit-org-form-organization-phone' placeholder='Organization phone number'/>
					</div>
					<div className='form-element-container'>
						<label htmlFor='submit-org-form-organization-street-address'>Street address:</label>
						<input 
							id='submit-org-form-organization-street-address' 
							name='org_st_addr' 
							type='text' 
							required 
							aria-labelledby='submit-org-form-organization-street-address' 
							placeholder='Street address'
						/>
					</div>
					<div className='form-element-container'>
						<label htmlFor='submit-org-form-organization-city'>City:</label>
						<input 
							id='submit-org-form-organization-city' 
							name='org_city' 
							type='text' 
							required 
							aria-labelledby='submit-org-form-organization-city' 
							placeholder='City'
						/>
					</div>
					<div className='form-element-container'>
						<label htmlFor='submit-org-form-organization-state'>State:</label>
						<select id='submit-org-form-organization-state' name='org_state' required aria-labelledby='submit-org-form-organization-state'>
							<option value='AL'>AL</option>
							<option value='AK'>AK</option>
							<option value='AZ'>AZ</option>
							<option value='AR'>AR</option>
							<option value='CA'>CA</option>
							<option value='CO'>CO</option>
							<option value='CT'>CT</option>
							<option value='DE'>DE</option>
							<option value='FL'>FL</option>
							<option value='GA'>GA</option>
							<option value='ID'>ID</option>
							<option value='HI'>HI</option>
							<option value='IL'>IL</option>
							<option value='IN'>IN</option>
							<option value='IA'>IA</option>
							<option value='KS'>KS</option>
							<option value='KY'>KY</option>
							<option value='LA'>LA</option>
							<option value='ME'>ME</option>
							<option value='MD'>MD</option>
							<option value='MA'>MA</option>
							<option value='MI'>MI</option>
							<option value='MN'>MN</option>
							<option value='MS'>MS</option>
							<option value='MO'>MO</option>
							<option value='MT'>MT</option>
							<option value='NE'>NE</option>
							<option value='NV'>NV</option>
							<option value='NH'>NH</option>
							<option value='NJ'>NJ</option>
							<option value='NM'>NM</option>
							<option value='NY'>NY</option>
							<option value='NC'>NC</option>
							<option value='ND'>ND</option>
							<option value='OH'>OH</option>
							<option value='OK'>OK</option>
							<option value='OR'>OR</option>
							<option value='PA'>PA</option>
							<option value='RI'>RI</option>
							<option value='SC'>SC</option>
							<option value='SD'>SD</option>
							<option value='TN'>TN</option>
							<option value='TX'>TX</option>
							<option value='UT'>UT</option>
							<option value='VT'>VT</option>
							<option value='VA'>VA</option>
							<option value='WA'>WA</option>
							<option value='WV'>WV</option>
							<option value='WI'>WI</option>
							<option value='WY'>WY</option>
						</select>
					</div>
					<div className='form-element-container'>	
						<label htmlFor='submit-org-form-organization-zip-code'>Zip code:</label>
						<input id='submit-org-form-organization-zip-code' name='org_zipcode' type='text' required aria-labelledby='submit-org-form-organization-zip-code' placeholder='Zip code'/>
					</div>
					<div className='form-element-container'>
						<label htmlFor='submit-org-form-organization-type'>Organization Type:</label>
						<select id='submit-org-form-organization-type' name='org_type' required aria-labelledby='submit-org-form-organization-type'>
							<option value='employment' disabled>Employment</option>
							<option value='housing'>Housing</option>
						</select>
					</div>
					<div className='form-error-container'></div>
					<div className='form-element-submit-container'>
						<input id='form-submit' type='submit' required aria-label='form-submit' value='Submit'/>
					</div>
				</fieldset>
			</form>
		);
	}
	
	renderSubmitOrgLoginMessage(){
		return(<div id='endorsement-login-message'><h3>You must <Link to='/login'>log in</Link> to submit organizations</h3></div>);
	}
	
	handleSubmit=e=>{
		e.preventDefault()
		const {org_name,org_phone,org_st_addr,org_city,org_state,org_zipcode,org_type}=e.target;
		OrgApiService.postOrg(org_name.value,org_phone.value,org_st_addr.value,org_city.value,org_state.value,org_zipcode.value,org_type.value)
			.catch(this.state.setError);
	}
	
	render(){
		return(
			<section id='form-container'>
				{TokenService.hasAuthToken()?this.renderSubmitOrgForm():this.renderSubmitOrgLoginMessage()}
			</section>
		);
	}

}