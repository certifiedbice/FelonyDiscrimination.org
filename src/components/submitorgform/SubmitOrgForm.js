import React,{Component} from 'react';
import OrgApiService from '../../services/org-api-service'
import './SubmitOrgForm.css';

export default class SubmitOrgForm extends Component{
    state={
        setError:null,
    }
    handleSubmit=e=>{
        e.preventDefault()
        const {org_name,org_phone,org_st_addr,org_city,org_state,org_zipcode,org_type}=e.target;
        OrgApiService.postOrg(org_name.value,org_phone.value,org_st_addr.value,org_city.value,org_state.value,org_zipcode.value,org_type.value)
            .catch(this.state.setError);
    }
    render(){
        return(
            <section id="review-form-container">
                    <form id="review-form" name="review-form" aria-label="Report discrimination form" onSubmit={this.handleSubmit}>
                        <fieldset>
                            <legend><h2>Review an Organization</h2></legend>
                            <div id="review-form-instructions">
                                <p>
                                    We invite you to submit organizations into the database, where they can then be reviewed as positive or negative for discrimination. 
                                </p>
                                <p>
                                    We would like to thank you for taking the time to submit an entry, every entry is one step closer to a solution.
                                </p>
                            </div>
                            <div className="review-form-element-container">
                                <label>Organization name:</label>
                                <input className="review-form-element" name="org_name" id="review-form-organization-name" type="text" required aria-labelledby="review-form-organization-name" placeholder="Name"/>
                            </div>
                            <div className="review-form-element-container">
                                <label>Organization phone:</label>
                                <input className="review-form-element" name="org_phone" id="review-form-organization-phone" type="tel" aria-labelledby="review-form-organization-phone" placeholder="Organization phone number"/>
                            </div>
                            <div className="review-form-element-container">
                                <label>Organization street address:</label>
                                <input 
                                    className="review-form-element" 
                                    name="org_st_addr" 
                                    id="review-form-organization-street-address" 
                                    type="text" 
                                    required 
                                    aria-labelledby="review-form-organization-street-address" 
                                    placeholder="Street address"
                                />
                            </div>
                            <div className="review-form-element-container">
                                <label>Organization City:</label>
                                <input 
                                    className="review-form-element" 
                                    name="org_city" 
                                    id="review-form-organization-city" 
                                    type="text" 
                                    required 
                                    aria-labelledby="review-form-organization-city" 
                                    placeholder="City"
                                />
                            </div>
                            <div className="review-form-element-container">
                                <label>Organization State:</label>
                                <select className="review-form-element" name="org_state" id="review-form-organization-state" required>
                                    <option value="AL">AL</option>
                                    <option value="AK">AK</option>
                                    <option value="AZ">AZ</option>
                                    <option value="AR">AR</option>
                                    <option value="CA">CA</option>
                                    <option value="CO">CO</option>
                                    <option value="CT">CT</option>
                                    <option value="DE">DE</option>
                                    <option value="FL">FL</option>
                                    <option value="GA">GA</option>
                                    <option value="HI">HI</option>
                                    <option value="ID">ID</option>
                                    <option value="IL">IL</option>
                                    <option value="IN">IN</option>
                                    <option value="IA">IA</option>
                                    <option value="KS">KS</option>
                                    <option value="KY">KY</option>
                                    <option value="LA">LA</option>
                                    <option value="ME">ME</option>
                                    <option value="MD">MD</option>
                                    <option value="MA">MA</option>
                                    <option value="MI">MI</option>
                                    <option value="MN">MN</option>
                                    <option value="MS">MS</option>
                                    <option value="MO">MO</option>
                                    <option value="MT">MT</option>
                                    <option value="NE">NE</option>
                                    <option value="NV">NV</option>
                                    <option value="NH">NH</option>
                                    <option value="NJ">NJ</option>
                                    <option value="NM">NM</option>
                                    <option value="NY">NY</option>
                                    <option value="NC">NC</option>
                                    <option value="ND">ND</option>
                                    <option value="OH">OH</option>
                                    <option value="OK">OK</option>
                                    <option value="OR">OR</option>
                                    <option value="PA">PA</option>
                                    <option value="RI">RI</option>
                                    <option value="SC">SC</option>
                                    <option value="SD">SD</option>
                                    <option value="TN">TN</option>
                                    <option value="TX">TX</option>
                                    <option value="UT">UT</option>
                                    <option value="VT">VT</option>
                                    <option value="VA">VA</option>
                                    <option value="WA">WA</option>
                                    <option value="WV">WV</option>
                                    <option value="WI">WI</option>
                                    <option value="WY">WY</option>
                                </select>
                            </div>
                            <div className="review-form-element-container">	
                                <label>Organization Zip code:</label>
                                <input className="review-form-element" name="org_zipcode" id="review-form-organization-zip-code" type="text" required aria-labelledby="review-form-organization-zip-code" placeholder="Zip code"/>
                            </div>
                            <div className="review-form-element-container">
                                <label>Organization Type:</label>
                                <select className="review-form-element" name="org_type" id="review-form-organization-type" required>
                                    <option value="Employment" disabled>Employment</option>
                                    <option value="Housing">Housing</option>
                                </select>
                            </div>
                            <div className="review-form-error-container">
                            </div>
                            <div className="review-form-element-container">
                                <input id="review-form-submit" type="submit" required aria-label="Review form submit" value="Submit"/>
                            </div>
                        </fieldset>
                    </form>
                </section>
        );
    }
}