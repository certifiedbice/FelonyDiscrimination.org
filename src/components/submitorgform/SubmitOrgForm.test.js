import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SubmitOrgForm from './SubmitOrgForm';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<SubmitOrgForm/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<SubmitOrgForm/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});