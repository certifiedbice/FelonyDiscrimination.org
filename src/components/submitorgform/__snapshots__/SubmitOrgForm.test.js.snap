// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="form-container"
>
  <form
    aria-label="submit-org-form"
    id="submit-org-form"
    name="submit-org-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        <h2
          id="submit-org-h2"
        >
          Submit an Organization
        </h2>
      </legend>
      <div
        id="submit-org-form-instructions"
      >
        <p>
          We invite you to submit organizations into the database, where they can then be reviewed as positive or negative for discrimination.
        </p>
        <p>
          We would like to thank you for taking the time to submit an entry, every entry is one step closer to a solution.
        </p>
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-name"
        >
          Organization name:
        </label>
        <input
          aria-labelledby="submit-org-form-organization-name"
          id="submit-org-form-organization-name"
          name="org_name"
          placeholder="Name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-phone"
        >
          Phone:
        </label>
        <input
          aria-labelledby="submit-org-form-organization-phone"
          id="submit-org-form-organization-phone"
          name="org_phone"
          placeholder="Organization phone number"
          type="tel"
        />
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-street-address"
        >
          Street address:
        </label>
        <input
          aria-labelledby="submit-org-form-organization-street-address"
          id="submit-org-form-organization-street-address"
          name="org_st_addr"
          placeholder="Street address"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-city"
        >
          City:
        </label>
        <input
          aria-labelledby="submit-org-form-organization-city"
          id="submit-org-form-organization-city"
          name="org_city"
          placeholder="City"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-state"
        >
          State:
        </label>
        <select
          aria-labelledby="submit-org-form-organization-state"
          id="submit-org-form-organization-state"
          name="org_state"
          required={true}
        >
          <option
            value="AL"
          >
            AL
          </option>
          <option
            value="AK"
          >
            AK
          </option>
          <option
            value="AZ"
          >
            AZ
          </option>
          <option
            value="AR"
          >
            AR
          </option>
          <option
            value="CA"
          >
            CA
          </option>
          <option
            value="CO"
          >
            CO
          </option>
          <option
            value="CT"
          >
            CT
          </option>
          <option
            value="DE"
          >
            DE
          </option>
          <option
            value="FL"
          >
            FL
          </option>
          <option
            value="GA"
          >
            GA
          </option>
          <option
            value="HI"
          >
            HI
          </option>
          <option
            value="ID"
          >
            ID
          </option>
          <option
            value="IL"
          >
            IL
          </option>
          <option
            value="IN"
          >
            IN
          </option>
          <option
            value="IA"
          >
            IA
          </option>
          <option
            value="KS"
          >
            KS
          </option>
          <option
            value="KY"
          >
            KY
          </option>
          <option
            value="LA"
          >
            LA
          </option>
          <option
            value="ME"
          >
            ME
          </option>
          <option
            value="MD"
          >
            MD
          </option>
          <option
            value="MA"
          >
            MA
          </option>
          <option
            value="MI"
          >
            MI
          </option>
          <option
            value="MN"
          >
            MN
          </option>
          <option
            value="MS"
          >
            MS
          </option>
          <option
            value="MO"
          >
            MO
          </option>
          <option
            value="MT"
          >
            MT
          </option>
          <option
            value="NE"
          >
            NE
          </option>
          <option
            value="NV"
          >
            NV
          </option>
          <option
            value="NH"
          >
            NH
          </option>
          <option
            value="NJ"
          >
            NJ
          </option>
          <option
            value="NM"
          >
            NM
          </option>
          <option
            value="NY"
          >
            NY
          </option>
          <option
            value="NC"
          >
            NC
          </option>
          <option
            value="ND"
          >
            ND
          </option>
          <option
            value="OH"
          >
            OH
          </option>
          <option
            value="OK"
          >
            OK
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="PA"
          >
            PA
          </option>
          <option
            value="RI"
          >
            RI
          </option>
          <option
            value="SC"
          >
            SC
          </option>
          <option
            value="SD"
          >
            SD
          </option>
          <option
            value="TN"
          >
            TN
          </option>
          <option
            value="TX"
          >
            TX
          </option>
          <option
            value="UT"
          >
            UT
          </option>
          <option
            value="VT"
          >
            VT
          </option>
          <option
            value="VA"
          >
            VA
          </option>
          <option
            value="WA"
          >
            WA
          </option>
          <option
            value="WV"
          >
            WV
          </option>
          <option
            value="WI"
          >
            WI
          </option>
          <option
            value="WY"
          >
            WY
          </option>
        </select>
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-zip-code"
        >
          Zip code:
        </label>
        <input
          aria-labelledby="submit-org-form-organization-zip-code"
          id="submit-org-form-organization-zip-code"
          name="org_zipcode"
          placeholder="Zip code"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-element-container"
      >
        <label
          htmlFor="submit-org-form-organization-type"
        >
          Organization Type:
        </label>
        <select
          aria-labelledby="submit-org-form-organization-type"
          id="submit-org-form-organization-type"
          name="org_type"
          required={true}
        >
          <option
            disabled={true}
            value="employment"
          >
            Employment
          </option>
          <option
            value="housing"
          >
            Housing
          </option>
        </select>
      </div>
      <div
        className="form-error-container"
      />
      <div
        className="form-element-submit-container"
      >
        <input
          aria-label="form-submit"
          id="form-submit"
          required={true}
          type="submit"
          value="Submit"
        />
      </div>
    </fieldset>
  </form>
</section>
`;
